challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -9433,6 +9433,1079 @@
+        "    # Using the same starting symbol table",
+        "    for each index of the input sequence:",
+        "        output the symbol at that index of the symbol table",
+        "        move that symbol to the front of the symbol table",
+        "</pre>",
+        "<table border=1 style='border-collapse:collapse;' cellpadding=5>",
+        "<tr>",
+        "<th>Input</th>",
+        "<th>Output</th>",
+        "<th>SymbolTable</th>",
+        "</tr>",
+        "<tr>",
+        "<td><b>b</b>roood</td>",
+        "<td>1</td>",
+        "<td>'abcdefghijklmnopqrstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>b<b>r</b>oood</td>",
+        "<td>1 17</td>",
+        "<td>'bacdefghijklmnopqrstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>br<b>o</b>ood</td>",
+        "<td>1 17 15</td>",
+        "<td>'rbacdefghijklmnopqstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>bro<b>o</b>od</td>",
+        "<td>1 17 15 0</td>",
+        "<td>'orbacdefghijklmnpqstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>broo<b>o</b>d</td>",
+        "<td>1 17 15 0 0</td>",
+        "<td>'orbacdefghijklmnpqstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>brooo<b>d</b></td>",
+        "<td>1 17 15 0 0 5</td>",
+        "<td>'orbacdefghijklmnpqstuvwxyz'</td>",
+        "</tr>",
+        "</table>",
+        "Encoding the string of character symbols 'broood' using a symbol table of",
+        "the characters 'a'-to-'z'",
+        "Decoding the indices back to the original symbol order:",
+        "<table border=1 style='border-collapse:collapse;' cellpadding=5>",
+        "<tr>",
+        "<th>Input</th>",
+        "<th>Output</th>",
+        "<th>SymbolTable</th>",
+        "</tr>",
+        "<tr>",
+        "<td><b>1</b> 17 15 0 0 5</td>",
+        "<td>b</td>",
+        "<td>'abcdefghijklmnopqrstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>1 <b>17</b> 15 0 0 5</td>",
+        "<td>br</td>",
+        "<td>'bacdefghijklmnopqrstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>1 17 <b>15</b> 0 0 5</td>",
+        "<td>bro</td>",
+        "<td>'rbacdefghijklmnopqstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>1 17 15 <b>0</b> 0 5</td>",
+        "<td>broo</td>",
+        "<td>'orbacdefghijklmnpqstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>1 17 15 0 <b>0</b> 5</td>",
+        "<td>brooo</td>",
+        "<td>'orbacdefghijklmnpqstuvwxyz'</td>",
+        "</tr>",
+        "<tr>",
+        "<td>1 17 15 0 0 <b>5</b></td>",
+        "<td>broood</td>",
+        "<td>'orbacdefghijklmnpqstuvwxyz'</td>",
+        "</tr>",
+        "</table>",
+        "Write a function that takes two parameters. The first parameter is a boolean (True indicates encode operation and False indicates decode operation). The second parameter is data to be encoded/decoded. The function should return the encode//decode operation result."
```

Comment:
"Write a function that takes two parameters. The first parameter is a boolean. True indicates an encode operation and false indicates a decode operation. The second parameter is either a string or an array to be encoded or decoded, respectively. If it is a string to be encoded, return an array of numbers. If it's an array of numbers to be decoded, return a string."